### PR TITLE
Fix/prefab fixes

### PR DESCRIPTION
**Pull request by yalania**
- Fix for the transform doing wrong loading 
- Fix for the components not being correct indetified as part of the original, I needed to add clone for the base component too. You will need to remove the extra mesh (second one) wrongly added to the scene and save for the fix to work.
- I did some fixes in order to create a empty mesh component

### CODE REVIEW

File: Engine/Component/Component.cpp [hunk 1]
```diff
@@ -17,6 +17,10 @@ void Component::Load(const Config &config)
 	SpecializedLoad(config);
 }
 
+void Component::CloneBase(Component* component) const
+{
+	*component = *this;
+}
```
Comment (JorxSS) [inline Engine/Component/Component.cpp:23]: for what is used now this Clone Base function? What happened or what's the difference with the other Clone?

File: Engine/Component/ComponentScript.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@ Component* ComponentScript::Clone(bool original_prefab) const
 		created_component = App->scripts->CreateComponentScript();
 	}
 	*created_component = *this;
+	CloneBase(static_cast<Component*>(created_component));
```
Comment (JorxSS) [inline Engine/Component/ComponentScript.cpp:39]: this with scripts needs to be specialized.

**Review by JorxSS — COMMENTED** [review]
The prefabs save the current transform status. 
Also the extra Mesh Renderers after delete are not longer there.

Great job!!

File: Engine/Component/Component.cpp (hunk 1 above)
Comment (yalania) [reply Engine/Component/Component.cpp:23]: I need to clone also the base component parameters and for not oding just *component = *this; everywhere I created the function.

**Review by mariofv — APPROVED** [review]
Works as intended, thanks!

**Review by JorxSS — APPROVED** [review]

**Pull request merged**